.github/workflows/cmake-msys2.yml

```diff
@@ -0,0 +1,36 @@
+name: CMake-MSYS2
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  msys2-ucrt64:
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: msys2/setup-msys2@v2
+        with:
+          msystem: UCRT64
+          update: true
+          pacboy: >-
+            cmake:p
+            gcc:p
```

Comment:
You can change this to MSYS2 environment agnostic `cc:p` so adding e.g. CLANG64 or CLANGARM64 in the future to the `msystem` matrix would be seamless.